### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by rafecolton**

### CODE REVIEW

**Comment by meatballhat** [comment]
tl;dr :thumbsdown: :scream_cat:

I think this role should be able to assume that docker is already installed and fully configured.  I'm pretty against adding bootstrap-y things to this role as it's perhaps going to be the most frequently executed of any role.  Sorry :frowning:

**Comment by meatballhat** [comment]
lest I appear to be a hypocrite, I would very much like to remove the installation of `inotify-tools` from this role :smiley_cat:

**Comment by rafecolton** [comment]
So we're going to have to do a `docker login` or write a `.dockercfg` almost every single time we use this.  Where should this functionality live?  In the bootstrap role?  Definitely don't want to be copying/pasting it all over the roles, and I also don't think it should be part of the image.

**Comment by meatballhat** [comment]
This is very much a bootstrap concern, IMHO, so yes in the bootstrap role (which is private!  shhhhhhh!)  :smiley_cat:

I don't want to copy pasta this stuff around either.  For me, docker login is a one-time kinda thing, whereas this role is a several-times-per-day kinda thing.

**Comment by rafecolton** [comment]
Ok, I can add it to bootstrap.  Wasn't honestly sure where it should live, so that's valid.  There are some other changes in this pr that I'd like to keep, so I'll reopen once I've cleaned it up

**Comment by rafecolton** [comment]
@meatballhat bump, just reverted the one commit

File: tasks/main.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@
   register: modcloth_app_pull_container
 
 - name: stop app
-  service: name={{ modcloth_app_name }} state=stopped
+  service: name={{ modcloth_app_name }} state=restarted
```
Comment (meatballhat) [inline tasks/main.yml:26]: Why restart here instead of stop?

File: templates/upstart.conf.j2 [hunk 2]
```diff
@@ -21,7 +21,7 @@ script
     docker rm -f {{ modcloth_app_name }} || true
   fi
 
-  docker run \
+  docker run --rm \
```
Comment (meatballhat) [inline templates/upstart.conf.j2:16]: I thought that `--rm` was the default now (???)

Comment (rafecolton) [inline templates/upstart.conf.j2:16]: It's the default for `docker build` but not for `docker run` (and they do different things, of course)

File: tasks/main.yml (hunk 1 above)
Comment (rafecolton) [inline tasks/main.yml:26]: Because a full restart is really what we want here.

Comment (meatballhat) [inline tasks/main.yml:26]: That's true, but that's why there's a `state=started` task just below that fires whether or not the pull changed.

Comment (rafecolton) [inline tasks/main.yml:26]: But again, if `restarted` is the state you want, why wouldn't you just use that?

Comment (meatballhat) [inline tasks/main.yml:26]: Because I only want to stop the container if the pull resulted in a change, but always want to ensure that the final state is `started`.  This means that for hosts that are already updated, the deploy step completes more quickly.  I'm very much wanting to keep the deployment as fast as possible (within reason).

Comment (meatballhat) [inline tasks/main.yml:26]: This is a strategy I've been using for quite awhile, fwiw.

Comment (meatballhat) [inline tasks/main.yml:26]: Ooh, just remembered something.  An upstart restart does not necessarily mean that the upstart conf is re-read.  For all I know this has been fixed in 14.04, but that was one of the biggest reasons why I stopped using restart during deploys.

Comment (rafecolton) [inline tasks/main.yml:26]: I have never experienced that issue...as far as I know, restarts of upstart services work exactly as expected (including restart docker containers)

Comment (jszwedko) [inline tasks/main.yml:26]: The behavior Dan describes about restart is apparently by design http://upstart.ubuntu.com/cookbook/#restart We've seen it as well.

File: templates/upstart.conf.j2 (hunk 2 above)
Comment (meatballhat) [inline templates/upstart.conf.j2:16]: :thumbsup:

File: tasks/main.yml (hunk 1 above)
Comment (rafecolton) [inline tasks/main.yml:26]: @jszwedko Ansible doesn't use `initctl restart`, it uses `stop` followed by `start`

Comment (jszwedko) [inline tasks/main.yml:26]: Ah! :+1:

Comment (jszwedko) [inline tasks/main.yml:26]: Actually, I don't think that is true: https://github.com/ansible/ansible/blob/devel/library/system/service#L480

Comment (jszwedko) [inline tasks/main.yml:26]: Maybe I'm wrong though, the service module is hard to follow.

Comment (rafecolton) [inline tasks/main.yml:26]: Agreed that it's hard to follow.  Well fwiw I've been using `state=restarted` for a while and it works as expected.  And I really think we should be declarative about our intent

Comment (jszwedko) [inline tasks/main.yml:26]: Mmm, actually lower down it appears to use stop/start https://github.com/ansible/ansible/blob/devel/library/system/service#L780

Comment (rafecolton) [inline tasks/main.yml:26]: So can we merge this?
